python/dataproc_templates/pubsublite/pubsublite_to_gcs.py

```diff
@@ -0,0 +1,115 @@
+            dest=constants.PUBSUBLITE_TO_GCS_INPUT_SUBSCRIPTION,
+            required=True,
+            help='Pubsublite to GCS Input subscription name'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_GCS_WRITE_MODE}',
+            dest=constants.PUBSUBLITE_TO_GCS_WRITE_MODE,
+            required=False,
+            default=constants.OUTPUT_MODE_APPEND,
+            help=(
+                'Output write mode '
+                '(one of: append,overwrite,ignore,errorifexists) '
+                '(Defaults to append)'
+            ),
+            choices=[
+                constants.OUTPUT_MODE_OVERWRITE,
+                constants.OUTPUT_MODE_APPEND,
+                constants.OUTPUT_MODE_IGNORE,
+                constants.OUTPUT_MODE_ERRORIFEXISTS
+            ]
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_GCS_OUTPUT_LOCATION}',
+            dest=constants.PUBSUBLITE_TO_GCS_OUTPUT_LOCATION,
+            required=True,
+            help='GCS output Bucket URL'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_CHECKPOINT_LOCATION}',
+            dest=constants.PUBSUBLITE_CHECKPOINT_LOCATION,
+            required=True,
+            help='Temporary folder for checkpoint location'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_GCS_OUTPUT_FORMAT}',
+            dest=constants.PUBSUBLITE_TO_GCS_OUTPUT_FORMAT,
+            required=False,
+            default=constants.FORMAT_CSV,
+            help=(
+                'Output Format to GCS '
+                '(one of: json, csv, avro, parquet) '
+                '(Defaults to csv)'
+            ),
+            choices=[
+                constants.FORMAT_AVRO,
+                constants.FORMAT_CSV,
+                constants.FORMAT_JSON,
+                constants.FORMAT_PRQT
+            ]
+        )
+
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        logger: Logger = self.get_logger(spark=spark)
+
+        # Arguments
+        input_subscription: str = args[constants.PUBSUBLITE_TO_GCS_INPUT_SUBSCRIPTION]
+        output_location: str = args[constants.PUBSUBLITE_TO_GCS_OUTPUT_LOCATION]
+        output_mode: str = args[constants.PUBSUBLITE_TO_GCS_WRITE_MODE]
+        pubsublite_checkpoint_location: str = args[constants.PUBSUBLITE_CHECKPOINT_LOCATION]
+        output_format: str = args[constants.PUBSUBLITE_TO_GCS_OUTPUT_FORMAT]
+
+        logger.info(
+            "Starting Pubsublite to GCS spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+
+        # Read
+        input_data=(spark.readStream \
+            .format(constants.FORMAT_PUBSUBLITE) \
+            .option(f"{constants.FORMAT_PUBSUBLITE}.subscription", input_subscription,) \
```

Comment:
How is this line working?
`.option(f"{constants.FORMAT_PUBSUBLITE}.subscription", input_subscription,)`

I think it should take the whole URL like: https://github.com/GoogleCloudPlatform/dataproc-templates/blob/4308d3a5a949404fa41c09ae21be7108ac0abc64/python/dataproc_templates/pubsublite/pubsublite_to_bigquery.py#L157

Can you please recheck?

Reply:
Yes, input_subscription is being passed in the above format only.
--pubsublite.to.gcs.input.subscription=projects/$GCP_PROJECT/locations/$REGION/subscriptions/test-sub-lite

The README also has the format mentioned for input_subscription.

Reply:
In that case, there is no problem and I see the README also instructs the user on how to provide this template parameter. 
From a purely readability point of view, we can rename the variable to `input_subscription_path`, since it contains the entire url as opposed to just the subscription name.

Reply:
Yes, Please go ahead. Also, do remove the extra , at last. I do not think it is required.

Reply:
Done.